From 441b39761c8a8d8b452c78f5baa272450a342b5d Mon Sep 17 00:00:00 2001
From: Alejandro Abdelnur <tucu@cloudera.com>
Date: Thu, 14 Mar 2013 17:46:59 -0700
Subject: [PATCH 0962/1357] HADOOP-9406. hadoop-client leaks dependency on JDK tools jar. (tucu)

git-svn-id: https://svn.apache.org/repos/asf/hadoop/common/trunk@1456729 13f79535-47bb-0310-9956-ffa450edef68
(cherry picked from commit 484217c9f3028100b57e99d20f241dc8442ddc75)

Conflicts:
	hadoop-common-project/hadoop-common/CHANGES.txt
---
 hadoop-client/pom.xml                            |   13 +++++++++++++
 hadoop-common-project/hadoop-annotations/pom.xml |   20 ++++++++++++++++++++
 hadoop-project/pom.xml                           |    9 ---------
 3 files changed, 33 insertions(+), 9 deletions(-)

diff --git a/hadoop-client/pom.xml b/hadoop-client/pom.xml
index a3586e8..dc0bd0b 100644
--- a/hadoop-client/pom.xml
+++ b/hadoop-client/pom.xml
@@ -293,6 +293,19 @@
         </exclusion>
       </exclusions>
     </dependency>
+
+    <dependency>
+      <groupId>org.apache.hadoop</groupId>
+      <artifactId>hadoop-annotations</artifactId>
+      <scope>compile</scope>
+      <exclusions>
+        <exclusion>
+          <groupId>jdk.tools</groupId>
+          <artifactId>jdk.tools</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+    
   </dependencies>
 
 </project>
diff --git a/hadoop-common-project/hadoop-annotations/pom.xml b/hadoop-common-project/hadoop-annotations/pom.xml
index 84f59ba..50a826a 100644
--- a/hadoop-common-project/hadoop-annotations/pom.xml
+++ b/hadoop-common-project/hadoop-annotations/pom.xml
@@ -35,4 +35,24 @@
     </dependency>
   </dependencies>
 
+  <profiles>
+    <profile>
+      <id>os.linux</id>
+      <activation>
+        <os>
+          <family>!Mac</family>
+        </os>
+      </activation>
+      <dependencies>
+        <dependency>
+          <groupId>jdk.tools</groupId>
+          <artifactId>jdk.tools</artifactId>
+          <version>1.6</version>
+          <scope>system</scope>
+          <systemPath>${java.home}/../lib/tools.jar</systemPath>
+        </dependency>
+      </dependencies>
+    </profile>
+  </profiles>
+
 </project>
diff --git a/hadoop-project/pom.xml b/hadoop-project/pom.xml
index 9b17b40..47fbe2c 100644
--- a/hadoop-project/pom.xml
+++ b/hadoop-project/pom.xml
@@ -927,15 +927,6 @@
       <properties>
         <build.platform>${os.name}-${os.arch}-${sun.arch.data.model}</build.platform>
       </properties>
-      <dependencies>
-        <dependency>
-          <groupId>jdk.tools</groupId>
-          <artifactId>jdk.tools</artifactId>
-          <version>1.6</version>
-          <scope>system</scope>
-          <systemPath>${java.home}/../lib/tools.jar</systemPath>
-        </dependency>
-      </dependencies>
     </profile>
     <profile>
       <id>os.mac</id>
-- 
1.7.0.4

